### DISCUSSION TITLE
External keycloak

### DISCUSSION THREAD

**Discussion by yetanothern** [apiman/apiman#1812] [Help] [answered]
Hello.
I have some difficulties to set up apiman on wildfly with external keycloak.
As mentioned in https://www.apiman.io/apiman-docs/installation-guide/2.2.0.Final/manager/production-manager.html I've deleted in **standalone-apiman.xml** all under /server/profile/subsystem "keycloak-server" except _\<web-context\>auth\</web-context\>_.
Then I've added in **standalone-apiman.xml** under /server/profile/subsystem "keycloak" the realm
```
<realm name="apiman">
    <realm-public-key>my-realm-key</realm-public-key>     
    <auth-server-url>http://<ip>:<port>/auth</auth-server-url>    
    <ssl-required>none</ssl-required>    
    <enable-cors>false</enable-cors>    
    <principal-attribute>preferred_username</principal-attribute>    
</realm>
```
Then I was trying to start wildfly with `./bin/standalone.sh -c standalone-apiman.xml` and got an error `FATAL [org.keycloak.services] (ServerService Thread Pool -- 76) Error during startup: java.lang.RuntimeException: Failed to load Keycloak configuration`.
What did I miss?

**Comment by msavy** [comment]
Probably want to delete: 
- modules/layers.conf (remove kc reference)
- modules/system/layers/keycloak
- bin/product.conf

We're no longer going to bundle KC and Apiman soon.

**Reply by yetanothern** [reply]
After doing so I got another stacktrace.
[external_keycloak_error.txt](https://github.com/apiman/apiman/files/8133319/external_keycloak_error.txt)

**Comment by yetanothern** [comment, marked as answer]
Managed to achieve that by commenting `<extension module="org.keycloak.keycloak-server-subsystem"/>` and merging provided section from guide `https://www.apiman.io/apiman-docs/installation-guide/2.2.0.Final/manager/production-manager.html#_configuring_keycloak_authentication_for_the_manager_api_and_ui` with secure-deployments under `<subsystem xmlns="urn:jboss:domain:keycloak:1.1">` in standalone-apiman.xml.

**Reply by volkflo** [reply]
@yetanothern Glad you solved the problem and thank you for sharing the solution here as well :)
